toolchain/check/import_cpp.cpp

```diff
@@ -89,6 +89,8 @@ static auto GenerateAst(Context& context, llvm::StringRef importing_file_path,
     context.emitter().Emit(loc, CppInteropParseWarning, num_warnings,
                            num_imports, diagnostics_str);
   }
+  // Remove link to the diagnostics consumer before its deletion.
+  ast->getDiagnostics().setClient(nullptr);
```

Comment:
Maybe use `llvm::make_scope_exit` up near where the ast is built, to ensure this happens even if early-returns are introduced to this function? (there's probably no useful named RAII class to use or create for this use case, so the one-off `make_scope_exit` seems OK here)

Reply:
oh, wait... is there a need for this to be delayed at all? Or could it be null'd out immediately after the `buildASTFromCodeWithARgs`?

Reply:
As suggested, I've moved this to be right after the `buildASTFromCodeWithARgs()` call.

Note though that while this works because we happen to not use the diagnostics consumer after that call, there's no reason to remove the consumer and if we change the code to use the AST after that point, it would not have a diagnostics consumer.
If the diagnostics consumer would have been kept for longer, there would not be a good reason to unlink it from the AST. The reason I add unlinking it from the AST is because we shouldn't keep a pointer to a deleted object, not because it's not used.